### ISSUE TITLE
Cannot launch Guard with 0.17.0pre version

### ISSUE THREAD

**Issue by ZipoKing** [celluloid/celluloid#619]
Cannot launch tests with Guard, but when launching them with rspec command then everything works OK.
All celluloid-related gems (-essentials, -extras, -fsm, -pool, -supervision) are taken from Github
Backtrace:
```
/usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-b6ca08ce22d8/lib/celluloid/calls.rb:48:in `check': wrong number of arguments (2 for 1) (ArgumentError)
        from /usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-b6ca08ce22d8/lib/celluloid/calls.rb:26:in `dispatch'
        from /usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-b6ca08ce22d8/lib/celluloid/call/sync.rb:16:in `dispatch'
        from /usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-b6ca08ce22d8/lib/celluloid/cell.rb:50:in `block in dispatch'
        from /usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-b6ca08ce22d8/lib/celluloid/cell.rb:76:in `block in task'
        from /usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-b6ca08ce22d8/lib/celluloid/actor.rb:363:in `block in task'
        from /usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-b6ca08ce22d8/lib/celluloid/task.rb:57:in `block in initialize'
        from /usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-b6ca08ce22d8/lib/celluloid/task/fibered.rb:14:in `block in create'
        from (celluloid):0:in `remote procedure call'
        from /usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-b6ca08ce22d8/lib/celluloid/call/sync.rb:45:in `value'
        from /usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-b6ca08ce22d8/lib/celluloid/proxy/sync.rb:40:in `method_missing'
        from /usr/lib/ruby/gems/2.2.0/gems/listen-2.9.0/lib/listen/listener.rb:215:in `_init_actors'
        from /usr/lib/ruby/gems/2.2.0/gems/listen-2.9.0/lib/listen/listener.rb:87:in `block in <class:Listener>'
        from /usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-fsm-41a9d1a2d553/lib/celluloid/fsm.rb:176:in `instance_eval'
        from /usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-fsm-41a9d1a2d553/lib/celluloid/fsm.rb:176:in `call'
        from /usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-fsm-41a9d1a2d553/lib/celluloid/fsm.rb:129:in `transition_with_callbacks!'
        from /usr/lib/ruby/gems/2.2.0/bundler/gems/celluloid-fsm-41a9d1a2d553/lib/celluloid/fsm.rb:97:in `transition'
        from /usr/lib/ruby/gems/2.2.0/gems/listen-2.9.0/lib/listen/listener.rb:100:in `start'
        from /usr/lib/ruby/gems/2.2.0/gems/guard-2.12.5/lib/guard/commander.rb:35:in `start'
        from /usr/lib/ruby/gems/2.2.0/gems/guard-2.12.5/lib/guard/cli/environments/valid.rb:16:in `start_guard'
        from /usr/lib/ruby/gems/2.2.0/gems/guard-2.12.5/lib/guard/cli.rb:113:in `start'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
        from /usr/lib/ruby/gems/2.2.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
        from /usr/lib/ruby/gems/2.2.0/gems/guard-2.12.5/lib/guard/aruba_adapter.rb:32:in `execute'
        from /usr/lib/ruby/gems/2.2.0/gems/guard-2.12.5/lib/guard/aruba_adapter.rb:19:in `execute!'
        from /usr/lib/ruby/gems/2.2.0/gems/guard-2.12.5/bin/_guard-core:11:in `<main>'
```

**Comment by ZipoKing** [comment]
@digitalextremist Is there any docs which may help with migrating other gems to 0.17.0 - I think that the problem is within listen gem which is not maintained now and I think that if I'll get some note about changes then I'll be able to make this fix by myself.

**Comment by digitalextremist** [comment]
@ZipoKing, `0.17.0` needs to be formally cut ( again ) and pushed to RubyGems ( again ) once feedback trickles in on the `github:` binding in `Gemfile` ... for that, you need `submodues: true` added, and to use the `0.17.0-prerelease` branch as I believe we talked about before.

With that done, as I see you did, this is the offender:

https://github.com/guard/listen/blob/v2.9.0/lib/listen/listener.rb#L215

There are several calls to `add` and `pool` which use the old API. /cc: @e2

**Comment by e2** [comment]
@ZipoKing - listen gem is still actively maintained (unofficially), though it isn't under active development

@digitalextremist - PRs are welcome (let me know if any Listen internals are causing trouble)

**Comment by digitalextremist** [comment]
@e2 sounds good, when I get a chance I'll submit a patch to be contingent on `0.17.0`

@ZipoKing I will keep this open until this specific issue is fixed at `listen`. Thank you for submitting this API compatibility issue caught in the wild.

**Comment by ZipoKing** [comment]
@digitalextremist @e2 Thanks lads!

**Comment by e2** [comment]
@ZipoKing - just released Guard 2.12.7, which supports Listen 3.0.0, which itself doesn't depend on Celluloid at all. This means it's up to the project to choose which Celluloid version to use - Listen/Guard should work either way.

@digitalextremist - you can close this.

**Comment by ZipoKing** [comment]
@e2 Happy to hear that! Thanks a million!
